### PR TITLE
Display all comments on an article

### PR DESCRIPTION
**Pull request by FreemindNation**
Please provide some feedback on this ticket

display comments on an article

### CODE REVIEW

File: src/components/Collapsible.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    const [isHidden, setIsHidden] = useState(false);
+
+    const toggleIsHidden = ()=> {
+        setIsHidden(!isHidden);
+    }
+
+    return (
+        <section>
+            <button onClick={toggleIsHidden}>
+                {isHidden ? 'Show' : 'Hide'} {contentDescriptor} ({commentCount})
```
Comment (nc-christian-dutton) [inline src/components/Collapsible.jsx:14]: Nice idea adding the commentCount, if I go to articles/37 which has no comments, I get visual feedback saying that there are no comments. Nice touch!

File: src/components/CommentsList.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    getCommentsByArticleId(article.article_id).then((res) => {
+      setComments(res.comments);
+      setIsLoading(false);
+    });
+  }, [article.article_id]);
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
```
Comment (nc-christian-dutton) [inline src/components/CommentsList.jsx:20]: Nice job with the loading pattern!

File: src/components/CommentsList.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          <button>Add comment</button>
+        </form>
+      </section>
+      <section>
+        {comments.map((comment) => {
+          return <CommentCard key={comment.comment_id} comment={comment} />;
+        })}
+      </section>
+    </section>
+  );
```
Comment (nc-christian-dutton) [inline src/components/CommentsList.jsx:36]: Really nice semantic HTML

**Comment by nc-christian-dutton** [comment]
Great job with this one! 

I have a more general suggestion, when I first load up your app you're telling the user to copy and paste the username in to the box. I guess this might be work in progress and you have an idea for this where people can select a user eventually? Fine if so. 

If not, I would just hard code the user that you plan to use anyway without asking the user to copy and paste it and remove the log in box. Just let them know who they are logged in as once they reach the home screen. Like I say you might have plans to change this anyway, but just in case!

**Comment by FreemindNation** [comment]
Thank you for the feedback Christian. The idea of the login page is to eventually have a guest account so a visitor can just click guest and go the home page.  it is more visual aesthetics than anything as i already have user hard coded in userContext.

**Pull request merged**